snakemake/utils.py

```diff
@@ -527,4 +527,31 @@ def argvquote(arg, force=True):
         return cmdline
 
 
+def os_sync():
+    """Ensure flush to disk"""
+    if not ON_WINDOWS:
+        os.sync()
+
+
+def find_bash_on_windows():
```

Comment:
Not sure I follow.  This is just a convenience function for locating a useable bash installation on windows. If we didn't have this every one who wants to run the test suite on windows needs to configure which bash to use depending on their setup. 

We can't just rely on finding `bash.exe` on the path since that defaults to the WSL bash (ubuntu) on new windows 10 system.